pkg/registry/apis/provisioning/jobs/export/resources.go

```diff
@@ -28,28 +32,70 @@ func ExportResources(ctx context.Context, options provisioning.ExportJobOptions,
 			return fmt.Errorf("get client for %s: %w", kind.Resource, err)
 		}
 
-		if err := exportResource(ctx, options, client, repositoryResources, progress); err != nil {
+		// When requesting v2 (or v0) dashboards over the v1 api, we want to keep the original apiVersion if conversion fails
+		var upgrader versionUpgrader
+		if kind.GroupResource() == resources.DashboardResource.GroupResource() {
+			var v2client dynamic.ResourceInterface
+			upgrader = func(ctx context.Context, item *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+				failed, _, _ := unstructured.NestedBool(item.Object, "status", "conversion", "failed")
+				if failed {
```

Comment:
you added new behavior but no test coverage :)

Reply:
the integration tests cover the expected behavior -- i am still skeptical on the specific approach/strategy and stubbing out registering a new v2 client felt like make work 😛 but yes if we think this is a reasonable long term strategy then worth it i suppose